diff --git a/mk.c b/mk.c
index 58379e8..956d6b5 100644
--- a/mk.c
+++ b/mk.c
@@ -204,8 +204,10 @@ int mk(char *argv[])
 		char *msg;
 		long msz;
 		mbox_get(mbox, mids[i], &msg, &msz);
-		printf("%c%c%03d", msg_stat(msg, msz, 0, 'N'),
-				msg_stat(msg, msz, 1, '0'), mids[i]);
+		// printf("%c%c%03d", msg_stat(msg, msz, 0, 'N'),
+		//		msg_stat(msg, msz, 1, '0'), mids[i]);
+		printf("%c%04d", msg_stat(msg, msz, 0, 'N'), mids[i]);
+
 		for (j = 0; ln[j]; j++) {
 			char *cln = ln[j];
 			char *tok = malloc(strlen(ln[j]) + 1);
